modules/collect/src/main/java/com/opengamma/collect/Guavate.java

```diff
@@ -389,4 +391,45 @@ private Guavate() {
         Collector.Characteristics.UNORDERED);
   }
 
+  /**
+   * Collector used at the end of a stream to build an immutable map
+   * from a stream containing map entries.
+   * <p>
+   * A collector is used to gather data at the end of a stream operation.
+   * This method returns a collector allowing streams to be gathered into
+   * an {@link ImmutableMap}.
+   * <p>
+   * This returns a map by converting each {@code Map.Entry} to a key and value.
+   * The input stream must resolve to unique keys.
+   * See {@link #toImmutableMap} for more details.
+   *
+   * @param <K> the type of the keys in the result map
+   * @param <V> the type of the values in the result map
+   * @return the immutable map collector
+   * @throws IllegalArgumentException if the same key is generated twice
+   */
+  public static <K, V> Collector<Map.Entry<K, V>, ?, ImmutableMap<K, V>> entriesToImmutableMap() {
+    return toImmutableMap(Map.Entry::getKey, Map.Entry::getValue);
+  }
+
+  /**
+   * Collector used at the end of a stream to build an immutable map
+   * from a stream containing pairs.
+   * <p>
+   * A collector is used to gather data at the end of a stream operation.
+   * This method returns a collector allowing streams to be gathered into
+   * an {@link ImmutableMap}.
+   * <p>
+   * This returns a map by converting each stream element to a key and value.
+   * The input stream must resolve to unique keys.
+   * See {@link Collectors#toMap(Function, Function)} for more details.
```

Comment:
One method has a see `toImmutableMap`, the other see `Collectors.toMap` and I can't see why